charts/agent/templates/_helpers.tpl

```diff
@@ -137,18 +137,16 @@ Daemonset labels
 {{- end -}}
 
 {{/*
-Use like: {{ include "get_or_fail_if_in_settings" (dict "root" . "key" "<mypath.key>" "setting" "<agent_setting>") }}
-Return the value of key "<mypath.key>" and if "<agent_setting>" is also defined in sysdig.settings.<agent_setting>, and error is thrown
-NOTE: I don't like the error message! Too much information.
+Use like: {{ include "get_if_not_in_settings" (dict "root" . "default" "<coalesced value>" "setting" "<agent_setting>") }}
+Return the default only if the value is not defined in sysdig.settings.<agent_setting>.
 */}}
-{{- define "get_or_fail_if_in_settings" -}}
-{{- $keyValue := tpl (printf "{{- .Values.%s -}}" .key) .root }}
-{{- if $keyValue -}}
-    {{- if hasKey .root.Values.sysdig.settings .setting }}{{ fail (printf "Value '%s' is also set via .sysdig.settings.%s'." .key .setting) }}{{- end -}}
-    {{- $keyValue -}}
+{{- define "get_if_not_in_settings" -}}
+{{- if not (hasKey .root.Values.sysdig.settings .setting) -}}
```

Comment:
I found this to be easier to use when coalescing multiple values, but got rid of the fail condition. Can think of a different approach if that kind of message is something we want to keep.